lib/src/currency_list_view.dart

```diff
@@ -119,90 +125,53 @@ class _CurrencyListViewState extends State<CurrencyListView> {
           ),
         ),
         Expanded(
-          child: ListView(
-            children: [
-              if (_favoriteList != null) ...[
-                ..._favoriteList!
-                    .map<Widget>((currency) => _listRow(currency))
-                    .toList(),
-                const Padding(
-                  padding: EdgeInsets.symmetric(horizontal: 20.0),
-                  child: Divider(thickness: 1),
-                ),
-              ],
-              ..._filteredList
-                  .map<Widget>((currency) => _listRow(currency))
-                  .toList()
-            ],
+          child: ListView.builder(
+            physics: widget.physics,
+            controller: widget.controller,
+            itemCount: _filteredList.length,
+            itemBuilder: (BuildContext context, int index) {
+              if (_favoriteList != null && index == 0) {
+                // list of favorite currencies
+                return Column(
+                  children: [
+                    ..._favoriteList!
+                        .map<Widget>((currency) => _listRow(currency))
+                        .toList(),
+                    const Padding(
+                      padding: EdgeInsets.symmetric(horizontal: 20.0),
+                      child: Divider(thickness: 1),
+                    ),
+                  ],
+                );
+              }
+              // list of currencies
+              return _listRow(_filteredList.elementAt(index));
+            },
```

Comment:
If the `_favoriteList` is not null, then the first currency from the `_filteredList` will not be displayed.

Reply:
New commit fix it. Maybe not pretty, but it works now